### PR TITLE
boards: nucleo_l552ze_q: do not offset the image by default by 0x400.

### LINKED ISSUE #28628: samples/tfm_integration/tfm_ipc: regression on nucleo_l552_ze
**Describe the bug**
Following merge of #28573, `samples/tfm_integration/tfm_ipc` is now failing on on nucleo_l552_ze, due to ns image offset mismatch.

**To Reproduce**
1. west build -b nucleo_l552ze_q_ns samples/tfm_integration/tfm_ipc/
2../build/nucleo_l552ze_q_ns/samples/tfm_integration/tfm_ipc/tfm/install/regression.sh
3. ./build/nucleo_l552ze_q_ns/samples/tfm_integration/tfm_ipc/tfm/install/TFM_UPDATE.sh
4. See error: TFM doesn't jump in ns image

**Expected behavior**
TFM should launch NS image

### PR DESCRIPTION
**Pull request by ioannisg**
We only need to offset the start of the non-secure image
by 0x400, if TFM is built with BL2 support. In this case
we use the ROM_START_OFFSET Kconfig switch and we set it
to 0x400, which is the default BL2 Header size value in
TF-M builds. This instructs the linker to offset the beginning
of the ROM section by 0x400. In other words, we do not need
to statically move the start of the image by 0x400 (using
FLASH_BASE_ADDRESS or FLASH_LOAD_OFFSET) when building for
nucleo_l552ze_q 'non-secure'. This fixes an issue that
prevents from running Zephyr + TFM without BL2 support.

Signed-off-by: Ioannis Glaropoulos <Ioannis.Glaropoulos@nordicsemi.no>

Fixes #28628

### CODE REVIEW

File: boards/arm/nucleo_l552ze_q/nucleo_l552ze_q_ns_defconfig [hunk 1]
```diff
@@ -50,5 +50,5 @@ CONFIG_CORTEX_M_SYSTICK=y
 CONFIG_RUNTIME_NMI=y
 CONFIG_TRUSTED_EXECUTION_NONSECURE=y
 # Flash configuration for TFM
-CONFIG_FLASH_LOAD_OFFSET=0x51400
-CONFIG_FLASH_LOAD_SIZE=0x2E000
+CONFIG_FLASH_LOAD_OFFSET=0x51000
```
Comment (ioannisg) [inline boards/arm/nucleo_l552ze_q/nucleo_l552ze_q_ns_defconfig:53]: @erwango in my opinion this is not a clean solution: we should not use the FLASH_LOAD_OFFSET to indicate the start of the non-secure Zephyr image. We should use the "non-secure" image partition and employ DT to extract the partition information. This is how it is done for MPS2, Musca and nRF platforms.

**Review by erwango — APPROVED** [review]
Tested successfully using `samples/tfm_integration/tfm_ipc/` on `nucleo_l552ze_q_ns`

File: boards/arm/nucleo_l552ze_q/nucleo_l552ze_q_ns_defconfig (hunk 1 above)
Comment (erwango) [reply boards/arm/nucleo_l552ze_q/nucleo_l552ze_q_ns_defconfig:53]: This is right. I'll do that, but will probably end up in V2.5.0 if not blocking.

Comment (microbuilder) [reply boards/arm/nucleo_l552ze_q/nucleo_l552ze_q_ns_defconfig:53]: @ioannisg Are you OK to merge this in for 2.4 as is if there isn't the bandwith to update quickly? https://github.com/zephyrproject-rtos/zephyr/pull/28573 was merged in too soon, and the changes should have gone in there.

Comment (ioannisg) [reply boards/arm/nucleo_l552ze_q/nucleo_l552ze_q_ns_defconfig:53]: Of course (that's why I opened this PR).

**Review by microbuilder — APPROVED** [review]

**Pull request merged**